### PR TITLE
Improve underline configuration

### LINKED ISSUE #423: Improve line configuration in tabs and navigation
In the Tabs component, the `lineHeight` accepts only the values of `borderBottomWidth`, while it should probably accept a `SizeConfig<number>` like the `activeVisualElement` of the Navigation component.

In the Tabs component, the `lineColor` accepts the values of "borderColor", while in the Navigation component the `lineColor` accepts the values of "background". We should probably unify them.

### PR DESCRIPTION
**Pull request by marcopiii**
Close #423

This PR is breaking since the Navigation's underline color is narrowed down from 
```
brandColor | backgroundColor | interactiveBackgroundColor | outlineColor | dataVisualizationColor 
```
to
```
brandColor | foregroundColor
```
and the Tab's config changes from
````
lineHeight: BentoSprinkles["borderBottomWidth"];
````
to 
```
lineHeight: SizeConfig<number>;
```

### CODE REVIEW

File: packages/bento-design-system/src/util/atoms.ts [hunk 1]
```diff
@@ -89,6 +89,7 @@ const color = {
   ...vars.interactiveForegroundColor,
   ...vars.outlineColor,
   ...vars.dataVisualizationColor,
+  transparent: "transparent",
```
Comment (marcopiii) [inline packages/bento-design-system/src/util/atoms.ts:92]: we had to add `transparent` to `color` to be able to pass such value to the navigation's `Box`

Comment (veej) [reply packages/bento-design-system/src/util/atoms.ts:92]: @gabro in order to use the a foreground color as a background, we're using the `background="currentColor"` trick. This means the decorator color is passed to the `color` Box prop, but decorator colors support the additional "transparent" value that is not supported by `color`. Thus, we added it to the available colors too, meaning that now it's theoretically possible to use "transparent" as a foreground color. Anything against it?

File: packages/bento-design-system/src/util/defaultConfigs.tsx [hunk 2]
```diff
@@ -483,7 +483,10 @@ export const underlineTabs: TabsConfig = {
     medium: 16,
     large: 32,
   },
-  lineHeight: 2,
+  lineHeight: {
+    medium: 2,
+    large: 4,
+  },
```
Comment (veej) [inline packages/bento-design-system/src/util/defaultConfigs.tsx:489]: is this a leftover, or is it deliberate?

**Review by veej — APPROVED** [review]
Just a doubt inline, and I'd wait @gabro's opinion on that "transparent" color

File: packages/bento-design-system/src/util/defaultConfigs.tsx (hunk 2 above)
Comment (marcopiii) [reply packages/bento-design-system/src/util/defaultConfigs.tsx:489]: deliberate: we want to be able to define different widths based on the size of the component.

Comment (gabro) [reply packages/bento-design-system/src/util/defaultConfigs.tsx:489]: It's ok to allow specifying a different height, but do we really want to change the value too? The more conservative option would be 
```suggestion
  lineHeight: {
    medium: 2,
    large: 2,
  },
```

**Review by gabro — COMMENTED** [review]
One question

**Review by gabro — APPROVED** [review]

**Pull request merged**